app/Models/Classes.php

```diff
@@ -177,6 +177,41 @@ public function baseSearch($search, $q = false)
         return $q;
     }
 
+    public function addClass($data)
+    {
+        if (empty($data['batch_id']) or empty($data['level_id'])) {
+            return false;
+        }
+
+        $batch_model = new Batch;
+        $batch = $batch_model->find($data['batch_id']);
+        $data['project_id'] = $batch->project_id;
+        $data['class_time'] = $batch->class_time;
+        
+        $class_day = $batch->day;
+        $class_time = $batch->class_time;
+
+        $day = strtolower(date('l', strtotime("Sunday + {$class_day} days")));
```

Comment:
Let people specify the date/time of the class as well. It can be an extra class that they are making - if so, it will be on a non standard time. IF no date is specified, use this logic - get the next class date. Otherwise, all good :-D